test/modules/post/test/all.rb

```diff
@@ -0,0 +1,155 @@
+require 'rex'
+
+lib = File.join(Msf::Config.install_root, "test", "lib")
+$LOAD_PATH.push(lib) unless $LOAD_PATH.include?(lib)
+require 'module_test'
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Test all applicable post modules',
+        'Description' => %q{ This module run all applicable post modules against the current session },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'alanfoster'],
+        'Platform' => [ 'linux', 'unix', 'osx', 'windows', 'java' ],
+        'SessionTypes' => [ 'meterpreter', 'shell', 'powershell' ]
+      )
+    )
+  end
+
+  def run
+    available_modules = select_available_modules
+    session_metadata = "#{session.session_type} session #{session.sid}"
+
+    print_status("Applicable modules:")
+    print_line(
+      matching_modules_table(available_modules, header: "Valid modules for #{session_metadata}", with_results: false)
+    )
+
+    module_results = run_modules(available_modules)
+
+    print_status("Modules results:")
+    print_line(matching_modules_table(module_results, header: "Results for #{session_metadata}", with_results: true))
```

Comment:
Should a failing wrapped module result in a `fail_with` here?

Reply:
This was aligning with the child test modules that don't report their error status currently

Was something like this in mind? I'm not sure if there's a better `fail_with` to use here

```ruby
    fail_count = module_results.sum { |result| result[:failures] }
    fail_with(Failure::BadConfig, "There were #{fail_count} failures") if fail_count > 0
```

<img width="861" alt="image" src="https://github.com/rapid7/metasploit-framework/assets/60357436/c1f7d245-a70a-4815-b30c-392cc40db165">

I'll treat this as not a blocker to not hold up the pipeline